codegen/src/main.rs

```diff
@@ -0,0 +1,268 @@
+
+                                let inner = activity
+                                    .result.ok_or_else(|| TurnkeyClientError::MissingResult)?
+                                    .inner.ok_or_else(|| TurnkeyClientError::MissingInnerResult)?;
+
+                                match inner {{
+                                    immutable_activity::result::Inner::{activity_result}(res) => Ok(res),
+                                    other => Err(TurnkeyClientError::UnexpectedInnerActivityResult(serde_json::to_string(&other)?))
+                                }}
+                                    
+                            }}
+                        "#
+                        )
+                    };
+
+                    println!("Generating {} (activity)", fn_name);
+                    generated_methods.push_str(&activity_func);
+                } else {
+                    let func = format!(
+                        r#"
+                        pub async fn {fn_name}(&self, request: coordinator::{req_type}) -> Result<coordinator::{res_type}, TurnkeyClientError> {{
+                            let url = format!("{{}}{{}}", self.base_url, "{route}");
+                            let post_body = serde_json::to_string(&request).unwrap();
+                            let stamp = stamp(post_body.clone(), &self.api_key).unwrap();
+
+                            let res = self.http
+                                .post(url)
+                                .header("X-Stamp", stamp)
+                                .body(post_body)
+                                .send()
+                                .await?;
+                            let parsed = res.json::<coordinator::{res_type}>().await?;
+                            Ok(parsed)
+                        }}
+                    "#
+                    );
+                    println!("Generating {} (query)", fn_name);
+                    generated_methods.push_str(&func);
+                }
+            }
+        }
+    }
+
+    // Insert the generated methods into the right spot inside of our template
+    let method_placeholder = "// <GENERATED_METHODS_PLACEHOLDER>";
+    let client_template = fs::read_to_string(CLIENT_TEMPLATE_PATH).unwrap();
+    let output = client_template.replace(method_placeholder, &generated_methods);
+
+    // Write the generated client to the `client` crate
+    let generated_client_path = PathBuf::from(GENERATED_CLIENT_DIR).join("client.rs");
+    fs::create_dir_all("../client/src/generated").unwrap();
+    fs::write(generated_client_path, output).expect("Failed to write generated file");
+
+    // Now add client.rs to the generated/mod.rs file
+    let mut mod_rs = fs::OpenOptions::new()
+        .append(true)
+        .open(out_dir.join("mod.rs"))
+        .unwrap();
+
+    // This will append a new line to the end of the file
+    writeln!(mod_rs, "\n// Added by tkhq_codegen").unwrap();
```

Comment:
It doesnt seem like this comment is in the mod.rs file, and the order of the `pub use` statements below doesnt seem to be preserved either for some reason?

I tried running `make generate` and see the same output, not a blocker or anything but just a bit confused

Reply:
Yeah this is because I'm post-processing mod.rs, which strips comments. I've fixed this now.